RecoCTPPS/PixelLocal/plugins/CTPPSPixelLocalTrackProducer.cc

```diff
@@ -0,0 +1,258 @@
+#include "DetectorDescription/Core/interface/DDRotationMatrix.h"
+#include "TMatrixD.h"
+#include "TVectorD.h"
+
+//------------------------------------------------------------------------------------------------//
+
+CTPPSPixelLocalTrackProducer::CTPPSPixelLocalTrackProducer(const edm::ParameterSet& parameterSet) :
+  parameterSet_(parameterSet){
+
+  inputTag_             = parameterSet_.getParameter<std::string>  ("label"                  );
+  verbosity_            = parameterSet_.getUntrackedParameter<int> ("RPixVerbosity"          );
+  maxHitPerRomanPot_    = parameterSet_.getUntrackedParameter<int> ("RPixMaxHitPerRomanPot"  );
+  maxHitPerPlane_       = parameterSet_.getUntrackedParameter<int> ("RPixMaxHitPerPlane"     );
+  maxTrackPerRomanPot_  = parameterSet_.getUntrackedParameter<int> ("RPixMaxTrackPerRomanPot");
+  maxTrackPerPattern_   = parameterSet_.getUntrackedParameter<int> ("RPixMaxTrackPerPattern" );
+  numberOfPlanesPerPot_ = parameterSet_.getUntrackedParameter<int> ("NumberOfPlanesPerPot"   );
+
+  std::string patterFinderAlgorithm = parameterSet_.getParameter<std::string>("RPixPatterFinderAlgorithm");
+  std::string trackFitterAlgorithm  = parameterSet_.getParameter<std::string>("RPixTrackFinderAlgorithm" );
+
+  // patter algorithm selector
+  if(patterFinderAlgorithm == "RPixRoadFinder"){
+   patternFinder_ = new RPixRoadFinder(parameterSet_);
+  }
+  else{
+    throw cms::Exception("CTPPSPixelLocalTrackProducer") << "Pattern finder algorithm" << patterFinderAlgorithm << " does not exist";
+  }
+  
+  for(uint32_t i=0; i<numberOfPlanesPerPot_; ++i){
+    listOfAllPlanes_.push_back(i);
+  }
+
+  //tracking algorithm selector
+  if(trackFitterAlgorithm == "RPixPlaneCombinatoryTracking"){
+    trackFinder_ = new RPixPlaneCombinatoryTracking(parameterSet_);
+  }
+  else{
+    throw cms::Exception("CTPPSPixelLocalTrackProducer") << "Tracking fitter algorithm" << trackFitterAlgorithm << " does not exist";
+  }
+  trackFinder_->setListOfPlanes(listOfAllPlanes_);
+  trackFinder_->initialize();
+  
+  tokenCTPPSPixelRecHit_ = consumes<edm::DetSetVector<CTPPSPixelRecHit> >(edm::InputTag(inputTag_));
+ 
+  produces<edm::DetSetVector<CTPPSPixelLocalTrack> > ();
+
+
+ 
+}
+
+//------------------------------------------------------------------------------------------------//
+
+CTPPSPixelLocalTrackProducer::~CTPPSPixelLocalTrackProducer(){
+  delete patternFinder_;
+  delete trackFinder_ ;
+}
+
+//------------------------------------------------------------------------------------------------//
+
+void CTPPSPixelLocalTrackProducer::fillDescriptions(edm::ConfigurationDescriptions & descriptions){
+  edm::ParameterSetDescription desc;
+
+  desc.add<std::string>    ("label"                            , "ctppsPixelRecHits"           );
+  desc.add<std::string>    ("RPixPatterFinderAlgorithm"        , "RPixRoadFinder"              );
+  desc.add<std::string>    ("RPixTrackFinderAlgorithm"         , "RPixPlaneCombinatoryTracking");
```

Comment:
is "RPix" prefix really useful?

capitalization style applies also to the config parameters